doc/ifa/SONiC_InbandFlowAnalyzer_FeatureSpec_Ver1.1.md

```diff
@@ -0,0 +1,478 @@
+                    "stage" : "INGRESS",
+                    "type" : "L3" ,
+                    "ports" : "Ethernet20"
+                    }
+            }
+
+            "ACL_RULE": {
+                "IFA|Flow1": {
+                    "PRIORITY" : "55",
+                    "IP_TYPE" : "ipv4any",
+                    "SRC_IP" : "10.10.0.26/32",
+                    "DST_IP" : "10.10.1.26/32",
+                    "PACKET_ACTION" : "int_insert"
+                }
+            }
+
+    ***Intermediate node***
+
+        IFA configuration
+            config ifa device-id 2346
+            config ifa device-type intermediate
+
+    ***Egress node***
+
+        IFA configuration
+            config ifa device-id 2347
+            config ifa collector-name collector1 ipv4 11.12.13.14 9070
+
+        ACL configuration
+            "ACL_TABLE": {
+                "IFA": {
+                    "policy_desc" : "IFA Egress Device Policy",
+                    "stage" : "INGRESS",
+                    "type" : "L3" ,
+                    "ports" : "Ethernet20"
+                    }
+            }
+
+            "ACL_RULE": {
+                "IFA|Flow1": {
+                    "PRIORITY" : "55",
+                    "IP_TYPE" : "ipv4any",
+                    "SRC_IP" : "10.10.0.26/32",
+                    "DST_IP" : "10.10.1.26/32",
+                    "PACKET_ACTION" : "int_delete"
+                }
+            }
+
+# 3 Design
+## 3.1 Overview
+
+***IFA Architecture***
+
+![IFA architecture](images/IFA_Arch.JPG)
+
+The above diagram illustrates the architecture of the IFA feature within SONiC.
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+
+IFA\_DEVICE\_TABLE
+
+    ;Defines IFA device configuration
+
+    key                = device         ; Only one instance and has a fixed key "device".
+    deviceid           = 1 * 5DIGIT     ; Uniquely identifies a device on the network to be analyzed
+    devicetype         = "intermediate" ; Optional parameter to specify device type as intermediate, value can be "intermediate" only.
+
+    Example:
+    127.0.0.1:6379[4]> keys *IFA_DEVICE*
+    1) "IFA_DEVICE_TABLE|device"
+
+    127.0.0.1:6379[4]> HGETALL "IFA_DEVICE_TABLE|device"
+    1) "deviceid"
+    2) 54325
+    3) "devicetype"
+    4) "intermediate"
+
+IFA\_COLLECTOR TABLE
+
+    ;Defines IFA collector configuration
+
+    key                 = name                      ; name is collector name and should be unique.
+    ipaddress-type      = "ipv4" / "ipv6"           ; Collector IP address type.
+    ipaddress           = ipv4_prefix / ipv6_prefix ; Collector IP address.
+    port                = 1 * 4DIGIT                ; Collector UDP port number
+
+    Example:
+    127.0.0.1:6379[4]> keys *IFA_COLLECTOR*
+    1) "IFA_COLLECTOR_TABLE|collector1"
+
+    127.0.0.1:6379[4]> HGETALL "IFA_COLLECTOR_TABLE|collector1"
+    1) "ipaddress-type"
+    2) "ipv4"
+    3) "ipaddress"
+    4) 10.20.30.40
+    5) "port"
+    6) 9070
+
```

Comment:
Is the communication to the collector from the switch is through the front panel routing interface or via management port?